### ISSUE TITLE
[Feature Request]: Ui suggestions about generate forever and hires fix

### ISSUE THREAD

**Issue by FloriMaster** [anapnoe/stable-diffusion-webui-ux#85]
### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits

### Generate forever:

The current way of accessing this feature is quite janky, especially on a mobile device where you have to long-press the Generate button and click on a tiny “Generate forever” button afterward.

Moreover, the current UI doesn’t allow the user to see if the “Generate forever” feature is currently enabled or disabled while generating images. It always shows “cancel generate forever” when you right-click on the “generate” button, regardless if the feature is on or off.

Therefore, I suggest adding a tick box next to the generate button instead of the right-click or long-press method.

### Hires fix:

After enabling the Hires fix, a lot of settings such as Hires steps, Upscaler, etc. appear and clutter up the UI. Typically, people don’t change these settings frequently. Unfortunately, there is no way to hide these settings without disabling Hires fix.

To improve this, I suggest adding an arrow button that will hide the settings, similar to what you did with the prompt box.

### Proposed workflow

.

### Additional information

![image](https://user-images.githubusercontent.com/125819629/234195088-f0db6723-6e89-4454-9ddc-edfdc8ab3991.png)
"Generate forever" button when generate forever is off (while generating images)

![2023-04-25_14-41](https://user-images.githubusercontent.com/125819629/234195352-12f792ad-8a4d-4536-a3bb-82e506069dfa.png)
"Generate forever" button when generate forever is on (while generating images)

![image](https://user-images.githubusercontent.com/125819629/234196722-a9c5c45d-44af-4adb-81d9-b9f11c0bbfb1.png)
something like this instead would be great

![image](https://user-images.githubusercontent.com/125819629/234198390-b67cc880-e6c8-451d-8526-86e65a8fa931.png)
something like this for the hires fix button.(i just changed html of the prompt box)

**Comment by Linaqruf** [comment]
Sorry for bringing up this issue. I thought it would be better to ask here instead of creating a new one. I have a problem with the 'generate forever' feature. Recently, my friend asked me about the implementation of 'generate forever,' and I remembered that it already existed. However, when I tried it, there was a bug and the right-click drop-down menu didn't appear correctly. Instead, it appeared under the footer.

![image](https://github.com/anapnoe/stable-diffusion-webui-ux/assets/50163983/2e569665-e0f4-4ac5-b759-e4dc9e80d743)

**Comment by anapnoe** [comment]
yes I am removing context menu and replace with friendly loop button just making the icon :)

**Comment by anapnoe** [comment]
I think interrupt serves no purpose when  toggle generate forever exists in the scenario that generate forever state is true

**Comment by SirVeggie** [comment]
Maybe the interrupt button can stop the infinite generation completely, and skip works as before? It's also fine as is to be honest.
Turning seed to random is annoying when you forget it, but it can also be useful to keep the same seed as you keep tweaking, so no need to change that behaviour.
A dropdown for hires fix might indeed be nice, maybe a like an arrow button next to the hires checkbox that will hide the hires. By default it's not hidden of course.
